Core/Src/canardRxQueue.c

```diff
@@ -0,0 +1,47 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <canard.h>
+#include "canardRxQueue.h"
+
+struct CanardRxQueueItem* rxQueueHead = NULL;
+struct CanardRxQueueItem* rxQueueTail = NULL;
+
+void enqueueRxFrame(CanardCANFrame* rx_frame) {
+
+  // dynamically allocate memory
+  struct CanardRxQueueItem* new = (struct CanardRxQueueItem*)malloc(sizeof(struct CanardRxQueueItem));
+  
+  if (new == NULL) {
+    printf("CanardRxQueueItem memory allocation failed");
+    return;
+  }
+
+  new->next = NULL;
+  new->frame = *rx_frame;
+
+  if (rxQueueTail) {
+    rxQueueTail->next = new;
+  } else {
+    rxQueueHead = new;
+  }
+
+  rxQueueTail = new;
+}
+
+
+struct CanardRxQueueItem* dequeueRxFrame() {
+
+  if (!rxQueueHead) {
+    return NULL; // queue is empty
+  }
+
+  struct CanardRxQueueItem* dequeuedItem  = rxQueueHead;
+  rxQueueHead = rxQueueHead->next;
+
+  if (!rxQueueHead) {
+    rxQueueTail = NULL; // after dequeuing, the queue is empty
+  }
```

Comment:
Since you dynamically allocated the queue item, the allocated memory should be handled carefully when we are done with it. You should really explicitly write a comment somewhere to let people know they need to free the memory block themselves after they pop it from the queue. Or, more preferably, you handle it for the user.